src/web/app/components/question-types/question-edit-answer-form/msq-question-edit-answer-form.component.ts

```diff
@@ -52,30 +59,27 @@ export class MsqQuestionEditAnswerFormComponent
   updateSelectedAnswers(index: number): void {
     this.isMsqOptionSelected[index] = !this.isMsqOptionSelected[index];
     this.disableNoneOfTheAboveOption();
+    const answersCopy: string[] = this.responseDetails.answers.slice();
     if (this.isMsqOptionSelected[index]) {
```

Comment:
Hmm, I use the updated values of `this.isMsqOptionSelected[index]` here and the `triggerResponseDetailsChange` is called later. So I removing `this.isMsqOptionSelected[index] = !this.isMsqOptionSelected[index];` as you suggested does not update the answers.

For instance, Say I am checking an option and the value changes from `false` to `true`. However on removing `this.isMsqOptionSelected[index] = !this.isMsqOptionSelected[index];` the value is not updated and `if (this.isMsqOptionSelected[index])` this will always result in false.